src/lib/Inference.hs

```diff
@@ -2248,102 +2313,109 @@ getSuperclassClosure givens newGivens = do
     alreadyVisited :: EnvReader m => Given n -> StateT1 Givens m n Bool
     alreadyVisited x = do
       Givens m <- get
-      ty <- getType x
+      ty <- getSynthType x
       return $ EKey ty `HM.member` m
 
     markAsVisited :: EnvReader m => Given n -> StateT1 Givens m n ()
     markAsVisited x = do
-      ty <- getType x
+      ty <- getSynthType x
       modify \(Givens m) -> Givens $ HM.insert (EKey ty) x m
 
 getDirectSuperclasses :: EnvReader m => Given n -> m n [Given n]
-getDirectSuperclasses dict = do
-  numSuperclasses <- getNumSuperclasses dict
+getDirectSuperclasses synthExpr = do
```

Comment:
You've defined a `SynthExpr` type above, but `synthExpr` here as a type of `Given`? I'm not sure how to interpret that

Reply:
Good point. Maybe we should remove the `Given` alias and use `SynthExpr` exclusively. "Given" refers to the role that a particular `SynthExpr` plays rather than the set of values it can take.

Reply:
I guess I was really just confused by the `synthExpr` name